src/helpers/MiddlewareEnv.js

```diff
@@ -0,0 +1,72 @@
+import fs from 'fs';
+import { URL } from "url";
+
+export class MiddlewareENV {
+  constructor() {
+    const {
+      PORT = "3000",
+      AGORIC_RPC = "http://0.0.0.0:26657",
+      FROM,
+      SUBMIT_RETRIES = "3",
+      SEND_CHECK_INTERVAL = "45",
+      BLOCK_INTERVAL = "6",
+      EI_CHAINLINKURL,
+      CREDENTIALS_FILE = "config/ei_credentials.json",
+      DB_FILE = "data/database.db",
+    } = process.env;
+
+    this.PORT = PORT;
+    this.AGORIC_RPC = AGORIC_RPC;
+    this.FROM = FROM;
+    this.SUBMIT_RETRIES = SUBMIT_RETRIES;
+    this.SEND_CHECK_INTERVAL = SEND_CHECK_INTERVAL;
+    this.BLOCK_INTERVAL = BLOCK_INTERVAL;
+    this.EI_CHAINLINKURL = EI_CHAINLINKURL;
+    this.CREDENTIALS_FILE = CREDENTIALS_FILE;
+    this.DB_FILE = DB_FILE;
+
+    this.validate();
+  }
+
+  /**
+   * This function validates the env vars
+   */
+  validate() {
+    assert(Number(this.PORT), "$PORT should be a valid number");
+    assert(Number(this.SUBMIT_RETRIES), "$SUBMIT_RETRIES should be a valid number");
+    assert(Number(this.SEND_CHECK_INTERVAL), "$SEND_CHECK_INTERVAL should be a valid number");
+    assert(Number(this.BLOCK_INTERVAL), "$BLOCK_INTERVAL should be a valid number");
```

Comment:
Number returns NaN if it's an invalid string. So better add a check !isNaN(Number(...)) here and for other number variables

Reply:
Fixed